### PR TITLE
Improvements, optimization, fixes

### PR DESCRIPTION
**Pull request by AkariDN**
Hello.
Thank you very much for implementing DRM support. Great job!

1. 0dd03f92005de5a0d8c5b4366531d0f721d3a4c3 8a2c6e3541d3abf8c7bf46c9cdb8a5e8b6a3d74a Improved stream parameters parsing:
   * Fix potential exceptions like re.search(...).group(1) when no matches found
   * Proper stream and DRM parameters parsing:
      * Don't search stream parameters in whole html, there can be multiple matches in other places
      * Get key url and headers from appropriate section (old version get them wherever finds)
      * Parse all headers, not only 'Authorization' (for futher potential site updates)
   * Support only 'hls' and 'dash' + 'widevine' for now, other types handled by inputstreamhelper
   * Pass 'license_type' and 'license_key' only if 'drm' section present
   * Auto-fallback to intermediate regex version in case of parsing errors
1. f9f5862e23e5726445a84c4d6b994140e8f9eee2 Play time progress synchonization with Wakanim can be switched on/off in plugin config. Localized strings review is needed.
1. 64481060c378f5ea1a577c3cb075768fdde13181 Do not load html twice everytime plugin starts (first time we load catalog in login() to check user logged in and the second one is our url). Instead, load our url and if not logged in - do login.

Updated 05.02.2018: all done

### CODE REVIEW

**Comment by AkariDN** [comment]
Wait please one more commit... Testing in progress.

File: resources/lib/streamparams.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    except:
+        return None
+
+
+def getStreamParams(args, s):
+    """Get stream parameters and check with inputstreamhelper
+    """
+    try:
+        r = get_stream_params_from_json(parse_stream_config(s.replace("autostart: (autoplay) ? \"true\" : \"false\"", "autostart: \"false\""), "jwplayer(\"jwplayer-container\").setup({"))
+    except:
```
Comment (MrKrabat) [inline resources/lib/streamparams.py]: statements like `except:` will not be accepted by the Kodi maintainers

File: resources/lib/streamparams.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        r['url'] = re.search(r"file:\s*(?P<q>['\"])(.+?)(?<!\\)(?P=q),", s).group(2)
+        r['proto'] = re.search(r"type:\s*(?P<q>['\"])(.+?)(?<!\\)(?P=q),", s).group(2)
+        if re.search(r"drm:\s*{", s):
+            r['drm'] = "widevine"
+            r['key'] = re.search(r"url:\s*(?P<q>['\"])(.+?)(?<!\\)(?P=q),", s).group(2)
+            r['headers'] = {"Authorization": re.search(r"value:\s*(?P<q>['\"])(.+?)(?<!\\)(?P=q)", s).group(2)}
+        else:
+            r['drm'] = None
+        return r
+    except:
```
Comment (MrKrabat) [inline resources/lib/streamparams.py]: statements like `except:` will not be accepted by the Kodi maintainers

File: resources/lib/streamparams.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
```
Comment (MrKrabat) [inline resources/lib/streamparams.py]: The whole code in this file looks like it was obfuscated, can you please rename the variables and add some comments.
Additionally we have enough disk space to not require one-line ifs and for-loops and you can add some blank lines for the readability.

No one will be able to maintain this code in the future.

File: resources/lib/streamparams.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        log(args, "Unknown stream protocol '%s'" % r['proto'], xbmc.LOGNOTICE)
+    if r['drm'] == "widevine":
+        r['drm'] = "com.widevine.alpha"
+    else:
+        log(args, "Unknown stream license type '%s'" % r['drm'], xbmc.LOGNOTICE)
+    try:
+        if not inputstreamhelper.Helper(r['proto'], r['drm']).check_inputstream():
+            log(args, "InputStreamHelper: check stream failed", xbmc.LOGERROR)
+            return None
+    except Exception, e:
```
Comment (MrKrabat) [inline resources/lib/streamparams.py]: I think catching the common `Exception` will also be declined by Team Kodi

**Comment by AkariDN** [comment]
Thanks for the comments, I'll fix it now.

**Comment by AkariDN** [comment]
Done.

Short multiline ifs, for-loops, long variable names, a lot of comments and other stuff breaks code visibility and increases development time. There are no special or complex in my code that cannot be understand or maintain, and we are not developing plugin for education purposes. Anyway it's my fault, I'll follow your and Kodi's maintainers coding style. Sorry for this.

**Pull request merged**